docs: add blog post for js-IPFS v0.39.0 release (#320)
ipfs/blog: content/post/072-js-ipfs-0.39.md

---
date: 2019-10-24
url: 071-js-ipfs-0-39
title: js-ipfs 0.39.0 released
author: Alex Potsides
---

# 🔦 Highlights

> > Configure your node with preset profiles! Limit outgoing browser requests! Start your node from fresh with one command!

Reviewer: Alan tends to do a truly delightful tweet embed. Totally optional - but would expand on the highlights a bit. =]

Author: I've reformatted it to be more similar to previous js-IPFS posts

```console
$ npm install -g ipfs
```

> ## 📋 Config profiles

Reviewer 2: There _must_ be a more exciting emoji!

```suggestion
## 🛠 Config profiles
```

Author: You're the expert 🤣

Author: It's all Greek to me: https://www.nytimes.com/interactive/2014/07/25/style/emoji-quiz.html

Author: discuss.ipfs.io does not share your emoji enthusiasm:

![image](https://user-images.githubusercontent.com/665810/67396951-633c9d00-f5a0-11e9-8a72-eb839a3558e5.png)

IPFS now supports the `jsipfs config profile` subcommand which you can use to update your configuration with certain presets suited for different use cases.

You can examine available profiles with the `jsipfs config profile ls` command:

```console
$ jsipfs config profile ls
server:
 Recommended for nodes with public IPv4 address (servers, VPSes, etc.)...
local-discovery:
 Sets default values to fields affected by `server` profile...
test:
 Reduces external interference, useful for running ipfs in test environments...
default-networking:
 Restores default network settings. Inverse profile of the `test` profile.
lowpower:
 Reduces daemon overhead on the system. May affect node functionality...
default-power:
 Inverse of "lowpower" profile.
 ```

Apply them with `jsipfs config profile apply`:

 ```console
$ jsipfs config profile apply lowpower
... output shows the difference between the old config and the new
 ```

> You may need to restart your daemon for changes to take effect.

Reviewer 2: ```suggestion
You will need to restart your daemon for changes to take effect.
```

Profiles can also be applied on init:

```console
$ jsipfs init --profile server
```

Or when starting the daemon (more on that in the 'Init and start' section below):

```console
$ jsipfs daemon --init-profile server
```

## 🌐 Concurrent HTTP requests limited in the browser

> Interacting with IPFS can cause a lot of preloading and other background requests to be made - most browsers can only have so many requests in flight to a single origin at once so this can cause requests to the same origins made outside of IPFS to take time as they wait their turn to be processed.

Reviewer 3: Perhaps we could link to some additional context, for example:
```suggestion
Interacting with IPFS can cause a lot of preloading and other background requests to be made - [most browsers](https://bugs.chromium.org/p/chromium/issues/detail?id=285567) can only have so many requests in flight to a single origin at once so this can cause requests to the same origins made outside of IPFS to take time as they wait their turn to be processed.
```

> As of `ipfs@0.39.0`, the concurrency of HTTP requests for DNS resolution and preloading are limited to always ensure there are always connections available to speedily process requests made by users.

Reviewer: ```suggestion
As of `ipfs@0.39.0`, the concurrency of HTTP requests for DNS resolution and preloading are limited to ensure there are always connections available to speedily process requests made by users.
```

## 🌅 Init and start your daemon in one command

When a new user first starts up the IPFS daemon they've been greeted by an unfriendly message telling them to `init` it first.

> No longer!  Now on first startup, the daemon will auto-init it's repo removing a tiny bit of friction from getting on to the distributed web.

Reviewer: ```suggestion
No longer!  Now on first startup, the daemon will auto-init its repo, removing a tiny bit of friction from getting on to the distributed web. 🙌
```

## 🚯 `block rm` now supported over the HTTP API

> Previously this command only worked with the daemon turned off, now you can use the semi-internal `jsipfs block rm` command with the daemon running.

Reviewer 2: ```suggestion
Previously this command only worked with the daemon turned off, now you can use the low level `jsipfs block rm` command with the daemon running.
```

> N.b. it's rare that you would need to use this command, a more common approach would be to call `jsipfs gc` instead which removes all non-pinned blocks from your IPFS repo.

Reviewer 2: ```suggestion
N.b. it's rare that you would need to use this command, a more common approach would be to call `jsipfs repo gc` instead which removes all non-pinned blocks from your IPFS repo.
```

> N.b. it's rare that you would need to use this command, a more common approach would be to call `jsipfs repo gc` instead which removes all non-pinned blocks from your IPFS repo.

Reviewer: what does n.b. mean? Not a common acronym AFAIK ;)

Author: [Nota bene](https://en.wiktionary.org/wiki/nota_bene) - means an aside or warning.  Is it not common?  🤷‍♂ I thought it was..

* [@vaultec81](https://github.com/vaultec81) (1 issue, 2 comments)
* [@vincepmartin](https://github.com/vincepmartin) (1 commit, 1 PR, 1 issue, 4 comments)
* [@vmx](https://github.com/vmx) (5 comments)
* [@yehia67](https://github.com/yehia67) (1 comment)
* [@Ziwei-Wei](https://github.com/Ziwei-Wei) (1 comment)

# 🙌🏽 Want to contribute?

Would you like to contribute to the IPFS project and don't know how? Well, there are a few places you can get started:

- Check the issues with the `help wanted` label in the [js-ipfs repo](https://github.com/ipfs/js-ipfs/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22)
- Join an IPFS All Hands, introduce yourself and let us know where you would like to contribute - https://github.com/ipfs/team-mgmt/#weekly-ipfs-all-hands
- Hack with IPFS and show us what you made! The All Hands call is also the perfect venue for demos, join in and show us what you built
- Join the discussion at http://discuss.ipfs.io/ and help users finding their answers.
- Join the [🚀 IPFS Core Implementations Weekly Sync 🛰](https://github.com/ipfs/team-mgmt/issues/992) and be part of the action!

# ⁉️ Do you have questions?

> The best place to ask your questions about IPFS, how it works and what you can do with it is at [discuss.ipfs.io](http://discuss.ipfs.io). We are also available at the `#ipfs` channel on Freenode.

Reviewer 3: ```suggestion
The best place to ask your questions about IPFS, how it works and what you can do with it is at [discuss.ipfs.io](https://discuss.ipfs.io). We are also available at the `#ipfs` channel on Freenode and [Matrix](https://matrix.to/#/#ipfs:matrix.org)
```
